kubernetes_state/datadog_checks/kubernetes_state/kubernetes_state.py

```diff
@@ -319,15 +319,18 @@ def _trim_job_tag(self, name):
     # Labels attached: namespace, pod
     # As a message the phase=Pending|Running|Succeeded|Failed|Unknown
     # From the phase the check will update its status
+    # Also submits as a gauge metric with tags so it is visualisable over time
     def kube_pod_status_phase(self, message, **kwargs):
         """ Phase a pod is in. """
+        metric_name = self.NAMESPACE + '.pod.status_phase'
         # Will submit a service check which status is given by its phase.
         # More details about the phase in the message of the check.
         check_basename = self.NAMESPACE + '.pod.phase'
         for metric in message.metric:
-            self._condition_to_tag_check(metric, check_basename, self.pod_phase_to_status,
-                                         tags=[self._label_to_tag("pod", metric.label),
-                                               self._label_to_tag("namespace", metric.label)] + self.custom_tags)
+            tags = [self._label_to_tag("pod", metric.label), self._label_to_tag("namespace", metric.label),
+                    self._label_to_tag("phase", metric.label)] + self.custom_tags
+            self._condition_to_tag_check(metric, check_basename, self.pod_phase_to_status, tags=tags)
+            self.gauge(metric_name, 1, tags)
```

Comment:
You're reporting the metric even if the value is `0` in kube-state-metrics. If the value is 0 we should report 0 as well, otherwise you get all pods in all phases all the time.

Check out this prometheus output for example:
```
kube_pod_status_phase{namespace="default",phase="Pending",pod="nginx-5876f47b65-992pl"} 0
kube_pod_status_phase{namespace="default",phase="Pending",pod="nginx-5876f47b65-kbxzm"} 0
kube_pod_status_phase{namespace="default",phase="Pending",pod="nginx-5876f47b65-wlrpr"} 0
kube_pod_status_phase{namespace="default",phase="Running",pod="nginx-5876f47b65-992pl"} 1
kube_pod_status_phase{namespace="default",phase="Running",pod="nginx-5876f47b65-kbxzm"} 1
kube_pod_status_phase{namespace="default",phase="Running",pod="nginx-5876f47b65-wlrpr"} 1
```